hadoop-hdfs-project/hadoop-hdfs-httpfs/src/test/java/org/apache/hadoop/lib/servlet/TestServerWebApp.java

```diff
@@ -43,6 +43,8 @@ public void getHomeDir() {
     assertEquals(ServerWebApp.getDir("TestServerWebApp0", ".log.dir", "/tmp/log"), "/tmp/log");
     System.setProperty("TestServerWebApp0.log.dir", "/tmplog");
     assertEquals(ServerWebApp.getDir("TestServerWebApp0", ".log.dir", "/tmp/log"), "/tmplog");
+    System.clearProperty("TestServerWebApp0.home.dir");
+    System.clearProperty("TestServerWebApp0.log.dir");
```

Comment:
@lzx404243 Nice catch!

If the test fails in L42, L43, or L45, `System.clearProperty` does not executed and the properties are not cleared. The property should be cleared in `@After` method or be cleared at the first of the test.

Reply:
Hi @aajisaka,
thanks for your review! You raised some good points.
Per further discussion with Xiaoqiao He in [jira link](https://issues.apache.org/jira/browse/HDFS-15691?focusedCommentId=17237118&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17237118), it looks like setting the system properties can be harmless as system properties are not reused in different tests.